api/test/e2e/consumer/consumer_test.go

```diff
@@ -44,57 +59,92 @@ var _ = Describe("Consumer", func() {
+			ExpectBody:   `"username":"consumer_1"`,
 		}),
-		Entry("update consumer", base.HttpTestCase{
+		Entry("List consumer (1 item)", base.HttpTestCase{
+			Object:       base.ManagerApiExpect(),
+			Method:       http.MethodGet,
+			Path:         "/apisix/admin/consumers",
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   `"total_size":1`,
+		}),
+		Entry("Create consumer #2", base.HttpTestCase{
 			Object: base.ManagerApiExpect(),
 			Method: http.MethodPut,
-			Path:   "/apisix/admin/consumers/consumer_1",
+			Path:   "/apisix/admin/consumers",
 			Body: `{
-				"username": "consumer_1",
+				"username": "consumer_2",
 				"plugins": {
 					"limit-count": {
 						"count": 2,
 						"time_window": 60,
-						"rejected_code": 504,
+						"rejected_code": 503,
 						"key": "remote_addr",
 						"policy": "local"
 					}
-				},
-				"desc": "test description"
+				}
 			}`,
 			Headers:      map[string]string{"Authorization": base.GetToken()},
 			ExpectStatus: http.StatusOK,
-			ExpectBody:   []string{"\"code\":0", "\"username\":\"consumer_1\"", "\"rejected_code\":504"},
+			ExpectBody:   `"username":"consumer_2"`,
 		}),
-		Entry("get consumer #2", base.HttpTestCase{
+		Entry("List consumer (2 item)", base.HttpTestCase{
 			Object:       base.ManagerApiExpect(),
 			Method:       http.MethodGet,
-			Path:         "/apisix/admin/consumers/consumer_1",
+			Path:         "/apisix/admin/consumers",
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   `"total_size":2`,
+		}),
+		Entry("List consumer (Paginate)", base.HttpTestCase{
+			Object:       base.ManagerApiExpect(),
+			Method:       http.MethodGet,
+			Path:         "/apisix/admin/consumers",
+			Query:        "page=2&page_size=1",
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   `"username":"consumer_2"`,
+		}),
+		Entry("Update consumer", base.HttpTestCase{
+			Object: base.ManagerApiExpect(),
+			Method: http.MethodPut,
+			Path:   "/apisix/admin/consumers/consumer_1",
+			Body: `{
+				"username": "consumer_1",
+				"plugins": {
+					"limit-count": {
+						"count": 2,
+						"time_window": 60,
+						"rejected_code": 504,
+						"key": "remote_addr",
+						"policy": "local"
+					}
+				}
+			}`,
 			Headers:      map[string]string{"Authorization": base.GetToken()},
 			ExpectStatus: http.StatusOK,
-			ExpectBody:   "\"rejected_code\":504",
+			ExpectBody:   []string{`"code":0`, `"username":"consumer_1"`, `"rejected_code":504`},
 		}),
-		Entry("delete consumer", base.HttpTestCase{
+		Entry("Batch Delete consumer", base.HttpTestCase{
 			Object:       base.ManagerApiExpect(),
 			Method:       http.MethodDelete,
-			Path:         "/apisix/admin/consumers/consumer_1",
+			Path:         "/apisix/admin/consumers/consumer_1,consumer_2",
```

Comment:
🤔 A little wired. One question: what will happen if deleting one of them failed?

Reply:
Good question.

Reply:
It performs deletion based on a for loop and returns an error when an error occurs, so when one of the multiple deletions is executed with an error, the subsequent deletions will be interrupted.
Since this does not cause unintended consequences (at most, some resources fail to be deleted, just try again), I don't think it needs to be modified.